garnish/system.py

```diff
@@ -49,11 +49,16 @@ def make_graph(self):
         for block_id, block in self.sys_dict['blocks'].items():
             moltype = block['moltype']
             n_mol = block['n_molecules']
-
-            n_at = self.sys_dict['topology'][moltype]['n_atoms']
-            a_types = self.sys_dict['topology'][moltype]['atomtypes']
-
-            connectivity = self.sys_dict['topology'][moltype]['connectivity']
+            try:
+                n_at = self.sys_dict['topology'][moltype]['n_atoms']
+                a_types = self.sys_dict['topology'][moltype]['atomtypes']
+
+                connectivity = self.sys_dict['topology'][moltype]['connectivity']
+            except KeyError:
+                # This fails when the molecule is not in the topology
+                # Instead of just breaking, i think it's better to continue but warn about it
+                print(f"Warning: Molecule {moltype} is in the structure, but not in the topology. Skipping...")
+                continue
```

Comment:
Uhm, I'm not sure this works properly... Doesn't it mess up completely the numbering? Any resulting structure is gonna be useless anyways, cause garnish has no clue which atom is which anymore.

Reply:
well that is why i asked.
handling the numbering shouldn't be too hard though.
Also i think the ignoring comments thing also works recursively on the itp files, so that would maybe also mess up numbering..